libbeat/cmd/instance/metrics.go

```diff
@@ -55,13 +59,29 @@ func setupMetrics(name string) error {
 		IncludeTop:   process.IncludeTopConfig{},
 	}
 	err := beatProcessStats.Init()
+	if err != nil {
+		return err
+	}
+
+	beatProcessSysInfo, err = sysinfo.Self()
```

Comment:
I'd strive for graceful degradation whereby if it cannot monitor a platform (e.g. `types.ErrNotImplemented` is returned) then don't report the associated metrics.